designs/node-ownership.md

```diff
@@ -0,0 +1,225 @@
+# Karpenter Node Ownership
+
+_Provisioning is used throughout this doc to reference Karpenter scheduling and spinning up a node. This can apply to both the pending pod scheduling loop and also the consolidation scheduling loop. Essentially, any scheduling loop where Karpenter launches pods._
+
+## 🔑  Summary
+
+Karpenter’s current node creation logic causes consistency issues as well as race conditions that lead to leaked instances and reduced node launch performance. With these changes, Karpenter will no longer create nodes, and instead allow the Kubelet to register and deregister the Node with the apiserver. As a result, Karpenter needs a new way to model in flight capacity: instances that have been launched, but have not yet been registered as a Node.
+
+The recommendation is that this in-flight capacity should be modeled as an internal Machine CR that represents a Karpenter provisioner/consolidation scheduling decision. This has the following benefits:
+
+1. Karpenter stops creating the Node object, avoiding race conditions and conflicts with the kube-apiserver and the kubelet that are causing orphaned nodes and instances
+2. Karpenter does not need to know the node name at the end of its provisioning loop, shortening the provisioning loop for clouds that do not know the node name at instance launch time
+3. Karpenter stops mutating the Node object by adding labels, ownerReferences and finalizers, avoiding conflicts with other Kubernetes tooling which relies on a well-known state for the Node.
```

Comment:
This should not really be a problem. It's perfectly fine for Karpenter to at least modify finalizers and labels. This should not interfere with other tooling. Given that Karpenter owns the label "namespace" and don't try to add/mutate labels owned by others.

Reply:
I can clarify in the doc. The bulk of the point here is that we can't Create the node with these values, which means they have to reconcile on. When these pieces are the basis for ownership and termination, this causes races where some dependency goes away before the reconciliation can occur (this is what is mentioned in the race condition section)